cpp/test/sg/dbscan_test.cu

```diff
@@ -46,72 +61,79 @@ template <typename T>
-    labels_ref_h.resize(len);
-    updateDevice(labels_ref, labels_ref_h.data(), params.n_row, stream);
 
-    T eps = 3.0;
-    int min_pts = 2;
-    cumlHandle handle;
-    handle.setStream(stream);
+    MLCommon::copy(labels_ref, l.data(), params.n_row, handle.getStream());
+
+    CUDA_CHECK(cudaStreamSynchronize(handle.getStream()));
+
+    dbscanFit(handle, out.data(), params.n_row, params.n_col, params.eps,
+              params.min_pts, labels, params.max_bytes_per_batch, false);
 
-    // forces a batch size of 2
-    size_t max_bytes_per_batch = 4 * sizeof(T) * 8;
+    CUDA_CHECK(cudaStreamSynchronize(handle.getStream()));
 
-    dbscanFit(handle, data, params.n_row, params.n_col, eps, min_pts, labels,
-              max_bytes_per_batch, true);
-    CUDA_CHECK(cudaStreamSynchronize(stream));
+    score = adjustedRandIndex(handle, labels_ref, labels, params.n_row, 0,
+                              params.n_centers - 1);
 
-    CUDA_CHECK(cudaStreamDestroy(stream));
+    if (score < 1.0) {
+      std::cout << "y: "
+                << arr2Str(labels_ref, 25, "labels_ref", handle.getStream())
+                << std::endl;
+      std::cout << "y_hat: "
+                << arr2Str(labels, 25, "labels", handle.getStream())
+                << std::endl;
+    }
   }
 
   void SetUp() override { basicTest(); }
 
   void TearDown() override {
-    CUDA_CHECK(cudaFree(data));
     CUDA_CHECK(cudaFree(labels));
     CUDA_CHECK(cudaFree(labels_ref));
   }
 
  protected:
   DbscanInputs<T> params;
-  T* data;
   int *labels, *labels_ref;
+
+  double score;
 };
 
 const std::vector<DbscanInputs<float>> inputsf2 = {
-  {0.05f, 6 * 2, 6, 2, 1234ULL}};
+  {50000, 16, 5, 0.01, 2, 2, (size_t)13e8, 1234ULL},
+  {500, 16, 5, 0.01, 2, 2, (size_t)100, 1234ULL},
+  {1000, 1000, 10, 0.01, 2, 2, (size_t)13e8, 1234ULL},
+  {50000, 16, 5, 0.01, 2, 2, (size_t)13e8, 1234ULL},
+  {20000, 10000, 10, 0.01, 2, 2, (size_t)13e8, 1234ULL},
+  {20000, 100, 5000, 0.01, 2, 2, (size_t)13e8, 1234ULL}};
 
 const std::vector<DbscanInputs<double>> inputsd2 = {
-  {0.05, 6 * 2, 6, 2, 1234ULL}};
+  {50000, 16, 5, 0.01, 2, 2, (size_t)13e9, 1234ULL},
+  {500, 16, 5, 0.01, 2, 2, (size_t)100, 1234ULL},
+  {1000, 1000, 10, 0.01, 2, 2, (size_t)13e9, 1234ULL},
+  {100, 10000, 10, 0.01, 2, 2, (size_t)13e9, 1234ULL},
+  {20000, 10000, 10, 0.01, 2, 2, (size_t)13e9, 1234ULL},
+  {20000, 100, 5000, 0.01, 2, 2, (size_t)13e9, 1234ULL}};
 
 typedef DbscanTest<float> DbscanTestF;
-TEST_P(DbscanTestF, Result) {
-  ASSERT_TRUE(devArrMatch(labels, labels_ref, params.n_row,
-                          CompareApproxAbs<float>(params.tolerance)));
-}
+TEST_P(DbscanTestF, Result) { ASSERT_TRUE(score == 1.0); }
 
 typedef DbscanTest<double> DbscanTestD;
-TEST_P(DbscanTestD, Result) {
-  ASSERT_TRUE(devArrMatch(labels, labels_ref, params.n_row,
-                          CompareApproxAbs<double>(params.tolerance)));
-}
+TEST_P(DbscanTestD, Result) { ASSERT_TRUE(score == 1.0); }
```

Comment:
Agreed with Thejaswi that it looks nice to have this score right in the c++ test!